main.tex

```diff
@@ -0,0 +1,175 @@
+
+\section{Ideation}
+
+Multiple ideas were assessed to absorb insights from the data:
+
+\begin{itemize}
+    \item \textbf{Service Level Distribution:} A count plot was used to examine the distribution of service levels across different orders.
+    \item \textbf{Daily Capacity Distribution:} A boxplot was created to explore the distribution of daily capacities across various warehouses.
+    \item \textbf{Weight vs. Shipping Cost:} A scatter plot was generated to understand how shipping costs vary with weight, colored by the carrier.
+\end{itemize}
+
+Additionally, a plot was generated to visualize the connections between \texttt{PlantCode} and \texttt{Port}, showing the logistics network and possible transportation routes between plants and ports.
+
+\section{Optimization Model}
+
+The optimization model was deployed as a MIP model. The objective was to minimize the total transportation cost by selecting the most cost-effective routes for delivering products from supply nodes (warehouses) to demand nodes (projects). However, one has to be sure about the passage through port 9, which is considered an obligatory constraint. 
+
+\subsection{Decision Variables}
+
+\begin{itemize}
+    \item \textbf{Route Decision Variables:} A binary decision variable was created for each route, showing whether a particular supply-to-demand route is chosen.
+\end{itemize}
+
+\subsection{Objective Function}
+
+The objective function is to minimize the total transportation cost, which is the sum of the transportation costs associated with each selected route. The objective function is given as:
+
+\[
+\text{Minimize } Z = \sum_{w \in W, b \in B} \text{cost}_{wb} \cdot x_{wb}
+\]
+
+Where:
+\begin{itemize}
+    \item \( \text{cost}_{wb} \) is the cost of transporting products from warehouse \( w \) to demand node \( b \),
+    \item \( x_{wb} \) is a binary variable indicating whether the route from warehouse \( w \) to demand node \( b \) is selected.
+\end{itemize}
+
+\subsection{Constraints}
+
+The optimization model is subject to several constraints:
+
+\begin{figure}[!h]
+     \begin{center}
+     \includegraphics[width=0.6\paperwidth]{route.png}
+          \end{center}
+     \caption{Route constraints}
+     \label{fig:cd}
+\end{figure}
+
+
+\begin{itemize}
+    \item \textbf{Supply Constraints:} The total quantity shipped from a warehouse cannot exceed its supply capacity.
+    \item \textbf{Demand Constraints:} The total quantity received at a demand node must meet or exceed its demand.
+    \item \textbf{Self-Transportation Constraints:} Routes from a supply node to itself were excluded.
+\end{itemize}
+
+\section{Results}
+
+
+
+The optimization was performed using the \texttt{Pulp} Python library, which solved the MIP problem using the default solver. The solution provided the optimal transportation routes and the associated costs. The following observations could be made:
```

Comment:
You should add a citation for the `Pulp` library to properly acknowledge its use. For example:

*"The optimization was performed using the `Pulp` Python library [1], which solved the MIP problem using the default solver."*  

Then, include the full citation in the references section, such as:  
```
[1] Stuart Mitchell et al., PuLP: A Linear Programming Toolkit for Python. Available at: https://coin-or.github.io/pulp/.
```
Additionally, briefly describe the solver used by `Pulp`. The default solver is CBC (Coin-or Branch and Cut), which is an open-source solver that uses branch-and-bound techniques to solve mixed-integer programming problems.